codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/config/TypeConstraint.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.config;
+
+/**
+ * Type Constraint for config value.
+ */
+public class TypeConstraint implements ConfigConstraint {
+
+    private final String type;
+
+    /**
+     * @param type ruby type for the config. Used for validation, must be a valid Ruby class.
```

Comment:
This is actually not exactly true - since Boolean isn't a Ruby class..... It certainly throws a wrench in how we do all of these.

I think lets just add "must be a valid Ruby class or `Boolean`".

Reply:
haha, that is very true. I'll update.